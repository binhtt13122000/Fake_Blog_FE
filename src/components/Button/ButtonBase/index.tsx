import * as React from "react";

import { Button as MaterialButton, ButtonProps as MaterialButtonProps } from "@material-ui/core";

export interface ButtonBaseProps extends MaterialButtonProps {}

const buttonStyle = {
    borderRadius: "calc(1rem + 6px)",
};

const ButtonBase = (props: ButtonBaseProps) => {
    return <MaterialButton {...props} style={buttonStyle} />;
};

export default ButtonBase;
