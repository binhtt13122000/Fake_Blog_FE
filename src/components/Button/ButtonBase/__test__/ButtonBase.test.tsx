import * as React from "react";

import ButtonBase, { ButtonBaseProps } from "../";

import { render, screen } from "@testing-library/react";

describe("<ButtonBase />", () => {
    const props: ButtonBaseProps = {
        children: <div>Test</div>,
    };

    beforeEach(() => {
        render(<ButtonBase {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });
});
