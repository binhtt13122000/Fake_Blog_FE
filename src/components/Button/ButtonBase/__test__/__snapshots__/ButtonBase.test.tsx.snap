// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBase /> should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    style="border-radius: calc(1rem + 6px);"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <div>
        Test
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
